regcomp.c

```diff
@@ -3310,7 +3328,9 @@ S_reg(pTHX_ RExC_state_t *pRExC_state, I32 paren, I32 *flagp, U32 depth)
             return ret;
         }
         else if (*RExC_parse == '?') { /* (?...) */
-            bool is_logical = 0;
+            bool is_logical, is_optimistic;
+          handle_qmark:
+            is_logical = 0;
```

Comment:
Why not put the label above the bool declarations so is_logical can be initialized in the same step?

You will need a `;` after it to give the label a statement, but I think it would be better.

The declaration for `is_optimistic` could be moved closer to first use too (immediately before the switch, I think).

Pretty much all of the declarations at the top of the block could be moved down (or out), that's outside the scope of the PR but it would let you avoid adding that semi-colon to the label.  But that's probably out of scope for this PR.

Reply:
I can see some other possible changes there, but they're unrelated to this PR.

Reply:
Please let me know what you suggest so I can queue up the second pr.